src/riscv/durable_storage/src/durable_storage/persistence_layer.rs

```diff
@@ -0,0 +1,88 @@
+// SPDX-FileCopyrightText: 2025 Trilitech <contact@trili.tech>
+//
+// SPDX-License-Identifier: MIT
+
+//! Implementation of the RISC-V PVM durable storage's persistence layer.
+//!
+//! A persistence layer is tied to a repository on disk, identified by a directory.
+//! Within that directory, the persistence layer needs to be able to perform basic KV operations:
+//! - get
+//! - set
+//! - delete
+//!
+//! As well as repository-level operations:
+//! - new
+//! - clone
+//! - commit (returning the commit hash)
+//! - checkout a specific commit
+
+use std::path::Path;
+
+/// Type alias for a 32-byte hash used for identifying key-value blobs & commits.
+type Hash = [u8; 32];
```

Comment:
Should we use the `blake3` hash type?

Reply:
I think this will make it easier for the merkle layer to construct the type needed for the `get` /  `delete` operations. The `blake3` is just a wrapper type on top of this precise array anyway, so it will just add an extra `.into()` at most call sites.

Reply:
I mean, having it be actually well typed would be nice? We have a lot of aliases in the code-base already, and in all honesty, it makes things harder to read sometimes